smalltalksrc/VMMakerTests/VMCogitHelpersTest.class.st

```diff
@@ -0,0 +1,128 @@
+Class {
+	#name : #VMCogitHelpersTest,
+	#superclass : #VMSimpleStackBasedCogitAbstractTest,
+	#instVars : [
+		'checkIsSmallInteger',
+		'checkNotSmallInteger'
+	],
+	#pools : [
+		'CogAbstractRegisters',
+		'CogRTLOpcodes'
+	],
+	#category : #'VMMakerTests-JitTests'
+}
+
+{ #category : #'as yet unclassified' }
+VMCogitHelpersTest >> assertIsInSmallIntegerRange: anInteger [
+
+	machineSimulator classRegisterValue: anInteger.
+
+	self runUntilReturnFrom: checkIsSmallInteger.
+	
+	self assert: machineSimulator classRegisterValue equals: 0
+]
+
+{ #category : #'as yet unclassified' }
+VMCogitHelpersTest >> assertNotInSmallIntegerRange: anInteger [
+
+	machineSimulator classRegisterValue: anInteger.
+
+	self prepareCall.
```

Comment:
I think this call kept old, you moved this line to `runUntilReturnFrom:` override method

Reply:
yes!

Reply:
```suggestion
```